cmd/snap/cmd_run.go

```diff
@@ -228,7 +228,11 @@ func isReexeced() bool {
 }
 
 func runSnapConfine(info *snap.Info, securityTag, snapApp, command, hook string, args []string) error {
-	snapConfine := filepath.Join(dirs.DistroLibExecDir, "snap-confine")
+	libExecDir := dirs.DistroLibExecDir
+	if isReexeced() {
```

Comment:
This should be done only once in this function. Right now it's being done twice, and it's also bogus because code right below will test for a path that does not exist. Proper way is likely to bring the code currently at the bottom here, and make sure snapConfine points to the proper thing upfront.